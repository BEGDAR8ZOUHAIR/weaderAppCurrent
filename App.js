import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import Home from './Screens/Home';

export default function App()
{
    return (
        <View style={styles.container}>
            <Home />
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
});



