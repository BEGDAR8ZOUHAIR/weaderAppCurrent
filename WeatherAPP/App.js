import { StyleSheet, Text, View } from 'react-native';

import Weather from './src/Weather';
export default function App() {
  
  return (
    <View style={styles.container}>
      <Weather />
    </View>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
});
