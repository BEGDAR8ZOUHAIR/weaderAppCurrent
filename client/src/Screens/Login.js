
import React, { useState, useEffect } from 'react';
import { View, TextInput,  Text, StyleSheet, TouchableOpacity } from 'react-native';

const Login = ({ navigation }) =>
{
  
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  const onSubmit = () =>
  {
    // handle login logic
    navigation.navigate('Home');
  };

   
  

  return (
    <View style={styles.container}>
      <Text style={styles.label}></Text>
      <TextInput
        style={styles.input}
        value={username}
        placeholder="Username"
        onChangeText={(username) => setUsername(username)}

      />
      <Text style={styles.label}></Text>
      <TextInput
        style={styles.input}
        secureTextEntry
        value={password}
        placeholder="Password"
        onChangeText={(password) => setPassword(password)}

      />
      <TouchableOpacity style={styles.button} onPress={onSubmit}>
        <Text style={{ color: 'white', fontSize: 18, textAlign: 'center' }}>Login</Text>
      </TouchableOpacity>
      
      {/* <Button title="Go to Home" onPress={() => navigation.navigate('Home')}></Button>
      <Button title="Go to Register" onPress={() => navigation.navigate('Register')}></Button> */}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#F5FCFF',
    padding: 30,

  },
  label: {
    fontSize: 18,
    marginTop: 10,
  },
  input: {
    width: '90%',
    borderColor: 'gray',
    borderWidth: 1,
    padding: 8,
    borderRadius: 5,
  },
  button: {
    backgroundColor: 'darkslateblue',
    marginTop: 15,
    color: 'tomato',
    paddingHorizontal: 30,
    paddingVertical: 10,
    borderRadius: 5
  },
});

export default Login;
