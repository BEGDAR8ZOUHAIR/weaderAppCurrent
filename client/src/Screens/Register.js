import React, { useState, useEffect } from 'react';
import { View, TextInput, Button, Text, StyleSheet ,TouchableOpacity } from 'react-native';

const Register = ({ navigation }) =>
{
  const [username, setUsername] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');


  const onSubmit = () =>
  {
    // handle registration logic
    navigation.navigate('Login');
  };

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Create an Account</Text>
      <Text style={styles.label}></Text>
      <TextInput
        style={styles.input}
        value={username}
        placeholder="Username"
        onChangeText={(text) => setUsername(text)}
      />
      <Text style={styles.label}></Text>
      <TextInput
        style={styles.input}
        value={email}
        placeholder="Email"
        onChangeText={(text) => setEmail(text)}
      />
      <Text style={styles.label}></Text>
      <TextInput
        style={styles.input}
        secureTextEntry
        value={password}
        placeholder="Password"
        onChangeText={(text) => setPassword(text)}
      />
      <Text style={styles.label}></Text>
      <TextInput
        style={styles.input}
        secureTextEntry
        value={confirmPassword}
        placeholder="Confirm Password"
        onChangeText={(text) => setConfirmPassword(text)}
      />
      <TouchableOpacity style={styles.button} onPress={onSubmit}>
        <Text style={{ color: 'white', fontSize: 18, textAlign: 'center' }}>Register</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#F5FCFF',
    padding: 30,

  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    textAlign: 'center',
  },
  label: {
    fontSize: 18,
    // margin: 10,

  },
  input: {
    height: 40,
    width: '90%',
    borderColor: 'gray',
    borderWidth: 1,
    marginTop: 8,
    padding: 8,
    borderRadius: 5,
    
  },
  button: {
    backgroundColor: 'darkslateblue',
    marginTop: 15,
    color: 'tomato',
    paddingHorizontal: 30,
    paddingVertical: 10,
    borderRadius: 5
  },
 
});

export default Register;
