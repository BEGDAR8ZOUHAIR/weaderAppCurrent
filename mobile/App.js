import React from 'react';
import { StyleSheet, Text, View, LogBox } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import Ionicons from 'react-native-vector-icons/Ionicons';
import Home from './Screens/Home';
const Stack = createNativeStackNavigator();
const Tab = createBottomTabNavigator();

const HomeStack = () =>
{
    LogBox.ignoreLogs(['Remote debugger']);

    return (
        <Stack.Navigator>
            <Stack.Screen name="Home" component={Home} />
        </Stack.Navigator>
    );
};


export default function App()
{
    return (
        <NavigationContainer>
            <Tab.Navigator
                screenOptions={({ route }) => ({
                    tabBarIcon: ({ focused, color, size }) =>
                    {
                        let iconName;

                        if (route.name === 'Home')
                        {
                            iconName = focused ? 'home' : 'home-outline';
                        } else if (route.name === 'Login')
                        {
                            iconName = focused ? 'person' : 'person-outline';
                        } else if (route.name === 'Register')
                        {
                            iconName = focused ? 'person-add' : 'person-add-outline';
                        }

                        return <Ionicons name={iconName} size={size} color={color} />;
                    },
                    tabBarActiveTintColor: 'darkslateblue',
                    tabBarInactiveTintColor: 'gray',
                })}
            >
                <Tab.Screen name="Home" options={{ headerShown: false }} component={Home} />
            </Tab.Navigator>
        </NavigationContainer>
    );
}
const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center'
    },
});

