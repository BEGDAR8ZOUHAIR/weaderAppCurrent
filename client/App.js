import { StyleSheet, Text, View } from 'react-native';
import React from 'react';

import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import Ionicons from 'react-native-vector-icons/Ionicons';

import Home from './src/Screens/Home';
import Login from './src/Screens/Login';
import Register from './src/Screens/Register';

const Stack = createNativeStackNavigator();
const Tab = createBottomTabNavigator();

  // options={{headerShown: false}}  after screen
const HomeStack = () =>

{
  return (
    <Stack.Navigator>
      <Stack.Screen  name="Login" component={Login} />
      <Stack.Screen  name="Home" component={Home} />
      <Stack.Screen name="Register" component={Register} />
    </Stack.Navigator>
  );
};


export default function App() {
  return (
    <NavigationContainer>
      <Tab.Navigator 
        screenOptions={({ route }) => ({
          tabBarIcon: ({ focused, color, size }) =>
          {
            let iconName;

            if (route.name === 'Home')
            {
            //  icons of page home 
              iconName = focused ? 'home' : 'home-outline';
            } else if (route.name === 'Login')
            {
              // icons of login
              iconName = focused ? 'person' : 'person-outline';
            } else if (route.name === 'Register')
            {
              // icons of register
              iconName = focused ? 'person-add' : 'person-add-outline';
            }

            // You can return any component that you like here!
            return <Ionicons name={iconName} size={size} color={color} />;
          },
          tabBarActiveTintColor: 'tomato',
          tabBarInactiveTintColor: 'gray',
        })}
      >
        <Tab.Screen name="Home" options={{ headerShown: false }} component={Home} />
        <Tab.Screen name="Login" component={Login} />
        <Tab.Screen name="Register" component={Register} />
      </Tab.Navigator> 
    </NavigationContainer>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent : 'center'
  },
});
 