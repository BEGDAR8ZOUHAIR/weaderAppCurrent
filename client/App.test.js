import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';
import Login from './src/Screens/Login';
import Register from './src/Screens/Register';


test('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
    const tree = renderer.create(<Register />).toJSON();
    expect(tree).toMatchSnapshot();
});
    





    