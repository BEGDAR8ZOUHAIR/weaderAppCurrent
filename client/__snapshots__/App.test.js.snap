// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenNavigationContainer
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RNSScreen
      activityState={2}
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          Array [
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            Array [
              Object {
                "flex": 1,
                "flexDirection": "column-reverse",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                  undefined,
                ]
              }
            />
            <RCTSafeAreaView>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginTop": 20,
                    "marginVertical": 50,
                    "position": "relative",
                  }
                }
              >
                <TextInput
                  onChangeText={[Function]}
                  placeholder="Search"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "borderRadius": 10,
                      "height": 50,
                      "marginBottom": 10,
                      "marginTop": 20,
                      "padding": 10,
                      "width": 300,
                    }
                  }
                  value="Safi"
                />
                <Text />
              </View>
            </RCTSafeAreaView>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenNavigationContainer>
  <View
    collapsable={false}
    onLayout={[Function]}
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "rgb(255, 255, 255)",
        "borderTopColor": "rgb(216, 216, 216)",
        "borderTopWidth": 0.5,
        "bottom": 0,
        "elevation": 8,
        "height": 49,
        "left": 0,
        "paddingBottom": 0,
        "paddingHorizontal": 0,
        "position": null,
        "right": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessibilityRole="tablist"
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="Home, tab, 1 of 3"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": true,
          }
        }
        accessibilityStates={
          Array [
            "selected",
          ]
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              Object {
                "color": "darkslateblue",
              },
              Object {
                "fontSize": 10,
              },
              undefined,
            ]
          }
        >
          Home
        </Text>
      </View>
      <View
        accessibilityLabel="Login, tab, 2 of 3"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": false,
          }
        }
        accessibilityStates={Array []}
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              Object {
                "color": "gray",
              },
              Object {
                "fontSize": 10,
              },
              undefined,
            ]
          }
        >
          Login
        </Text>
      </View>
      <View
        accessibilityLabel="Register, tab, 3 of 3"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": false,
          }
        }
        accessibilityStates={Array []}
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              Object {
                "color": "gray",
              },
              Object {
                "fontSize": 10,
              },
              undefined,
            ]
          }
        >
          Register
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`renders correctly 2`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenNavigationContainer
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RNSScreen
      activityState={2}
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          Array [
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            Array [
              Object {
                "flex": 1,
                "flexDirection": "column-reverse",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                  undefined,
                ]
              }
            />
            <RCTSafeAreaView>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginTop": 20,
                    "marginVertical": 50,
                    "position": "relative",
                  }
                }
              >
                <TextInput
                  onChangeText={[Function]}
                  placeholder="Search"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "borderRadius": 10,
                      "height": 50,
                      "marginBottom": 10,
                      "marginTop": 20,
                      "padding": 10,
                      "width": 300,
                    }
                  }
                  value="Safi"
                />
                <Text />
              </View>
            </RCTSafeAreaView>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenNavigationContainer>
  <View
    collapsable={false}
    onLayout={[Function]}
    pointerEvents="auto"
    style={
      Object {
        "backgroundColor": "rgb(255, 255, 255)",
        "borderTopColor": "rgb(216, 216, 216)",
        "borderTopWidth": 0.5,
        "bottom": 0,
        "elevation": 8,
        "height": 49,
        "left": 0,
        "paddingBottom": 0,
        "paddingHorizontal": 0,
        "position": null,
        "right": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessibilityRole="tablist"
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityLabel="Home, tab, 1 of 3"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": true,
          }
        }
        accessibilityStates={
          Array [
            "selected",
          ]
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              Object {
                "color": "darkslateblue",
              },
              Object {
                "fontSize": 10,
              },
              undefined,
            ]
          }
        >
          Home
        </Text>
      </View>
      <View
        accessibilityLabel="Login, tab, 2 of 3"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": false,
          }
        }
        accessibilityStates={Array []}
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              Object {
                "color": "gray",
              },
              Object {
                "fontSize": 10,
              },
              undefined,
            ]
          }
        >
          Login
        </Text>
      </View>
      <View
        accessibilityLabel="Register, tab, 3 of 3"
        accessibilityRole="button"
        accessibilityState={
          Object {
            "selected": false,
          }
        }
        accessibilityStates={Array []}
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "flex-end",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          >
            <Text />
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 25,
                  "position": "absolute",
                  "width": "100%",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text />
          </View>
        </View>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "textAlign": "center",
              },
              Object {
                "color": "gray",
              },
              Object {
                "fontSize": 10,
              },
              undefined,
            ]
          }
        >
          Register
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`renders correctly 3`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
      "padding": 30,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "borderRadius": 50,
        "height": 100,
        "width": 100,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginTop": 10,
      }
    }
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Username"
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 5,
        "borderWidth": 1,
        "padding": 8,
        "width": "90%",
      }
    }
    value=""
  />
  <Text
    style={
      Object {
        "fontSize": 18,
        "marginTop": 10,
      }
    }
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Password"
    secureTextEntry={true}
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 5,
        "borderWidth": 1,
        "padding": 8,
        "width": "90%",
      }
    }
    value=""
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "darkslateblue",
        "borderRadius": 5,
        "color": "tomato",
        "marginTop": 15,
        "opacity": 1,
        "paddingHorizontal": 30,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      Login
    </Text>
  </View>
</View>
`;

exports[`renders correctly 4`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
      "padding": 30,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Create an Account
  </Text>
  <Text
    style={
      Object {
        "fontSize": 18,
      }
    }
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Username"
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 40,
        "marginTop": 8,
        "padding": 8,
        "width": "90%",
      }
    }
    value=""
  />
  <Text
    style={
      Object {
        "fontSize": 18,
      }
    }
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Email"
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 40,
        "marginTop": 8,
        "padding": 8,
        "width": "90%",
      }
    }
    value=""
  />
  <Text
    style={
      Object {
        "fontSize": 18,
      }
    }
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Password"
    secureTextEntry={true}
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 40,
        "marginTop": 8,
        "padding": 8,
        "width": "90%",
      }
    }
    value=""
  />
  <Text
    style={
      Object {
        "fontSize": 18,
      }
    }
  />
  <TextInput
    onChangeText={[Function]}
    placeholder="Confirm Password"
    secureTextEntry={true}
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 5,
        "borderWidth": 1,
        "height": 40,
        "marginTop": 8,
        "padding": 8,
        "width": "90%",
      }
    }
    value=""
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "darkslateblue",
        "borderRadius": 5,
        "color": "tomato",
        "marginTop": 15,
        "opacity": 1,
        "paddingHorizontal": 30,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      Register
    </Text>
  </View>
</View>
`;
